lock-api-objects/src/main/java/com/palantir/lock/v2/TimelockService.java

```diff
@@ -110,4 +112,28 @@ default List<StartIdentifiedAtlasDbTransactionResponse> startIdentifiedAtlasDbTr
     void tryUnlock(Set<LockToken> tokens);
 
     long currentTimeMillis();
+
+    /**
+     * Acquires a lease on a named timestamp. The lease is taken out with a new fresh timestamp.
+     * The returned timestamps are fresh timestamps obtained strictly after the lease is taken out.
+     */
+    @RestrictedApi(
+            explanation =
+                    "This method is for internal Atlas and internal library use only. Clients MUST NOT use it unless"
+                        + " given explicit approval. Mis-use can result in SEVERE DATA CORRUPTION and the API contract"
+                        + " is subject to change at any time.",
+            allowlistAnnotations = ReviewedRestrictedApiUsage.class)
+    AcquireNamedMinTimestampLeaseResult acquireNamedMinTimestampLease(String timestampName, int numFreshTimestamps);
```

Comment:
@fsamuel-bs in your part, you will be forced to call this multiple times for EACH timestampName. This probably doesn't need to happen right now, but I imagine eventually, we want a batch version of this API? so you can grab multiple of these in a single RPC.

Reply:
Do we need that? I think the implementation will eventually use a disruptor such that users don't need to be aware of the batching at the API layer

Or, perhaps you are referring to something else